tests/ForEachExecutor.test.js

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable no-unused-vars */
+
+window.__DEV__ = true;
+window.__TEST__ = true;
+
+import TestUtils from 'react-addons-test-utils';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Ringa, {forEach, forEachParallel, event, __hardReset} from '../src/index';
+import TestController from './shared/TestController';
+import CommandSimple from './shared/CommandSimple';
+
+describe('ForEachExecutor', () => {
+  let command, domNode, reactNode, controller;
+
+  beforeEach(() => {
+    domNode = ReactDOM.findDOMNode(TestUtils.renderIntoDocument(
+      <div>Controller Attach Point</div>
+    ));
+
+    controller = new TestController('testController', domNode, {
+      timeout: 500
+    });
+
+  });
+
+  afterEach(() => {
+    __hardReset();
+  });
+
+  //----------------------------------------------------------------------
+  // forEach: Iterates over an array calling the executor for each
+  //----------------------------------------------------------------------
+  it('forEach: Iterates over an array calling the executor for each', (done) => {
```

Comment:
Instead of saying "forEach" in front of each test, wrap with a nested describe:
```
describe('forEach', () => {
    .....tests.....
});
```
See Model.test.js for an example.